cloudformation/jolly-roger.yaml

```diff
@@ -522,6 +555,27 @@ Resources:
                         "max-size": "100m"
                       }
                     }
+                - path: /usr/local/bin/renew-certificate
+                  content: |
+                    #!/bin/bash
+                    set -e
+
+                    # Renew the certificate and update haproxy's offline copy.
+                    if [ -n "${CertNotificationEmail}" ]; then
+                      email_flag="-m ${CertNotificationEmail}"
+                    else
+                      email_flag="--register-unsafely-without-email"
+                    fi
+                    certbot -v certonly --dns-route53 -d ${AppUrl} --non-interactive --agree-tos $email_flag
+                    mkdir -p /etc/haproxy/certs/
+                    cat /etc/letsencrypt/live/${AppUrl}/fullchain.pem /etc/letsencrypt/live/${AppUrl}/privkey.pem | tee /etc/haproxy/certs/cert.pem > /dev/null
+
+                    # If haproxy is running, perform a live certificate update.
+                    if [ -f /var/run/haproxy/stats.sock ]; then
+                      echo -e "set ssl cert /usr/local/etc/haproxy/certs/cert.pem <<\n$(cat /etc/haproxy/certs/cert.pem)\n" | socat stdio /var/run/haproxy/stats.sock
+                      echo -e "commit ssl cert /usr/local/etc/haproxy/certs/cert.pem" | socat stdio /var/run/haproxy/stats.sock
+                    fi
```

Comment:
Mildly curious if haproxy will reload the certs if you just send it a SIGHUP.

Reply:
I tried it and it did seem to reload them.

Here's more background on the mechanism: https://www.haproxy.com/blog/dynamic-ssl-certificate-storage-in-haproxy. Maybe slightly more complicated than it needs to be; I'm fine with doing something simpler if you'd rather, though this seemed like the most targeted approach and wasn't too bad to get working.

Reply:
Yeah I guess this is probably cleaner - especially in the presence of long-lived connections (which I think a SIGHUP might gracefully shutdown). It's also certainly useful to have the haproxy socket exposed for debugging/inspection purposes.